### ISSUE TITLE
Regent: Compiler error in bounds-checks mode

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#493]
When I try to compile the code in this tarball (run `./compile.sh`):

[bug-bounds.tar.gz](https://github.com/StanfordLegion/legion/files/2912802/bug-bounds.tar.gz)

I get multiple error messages like this:

```
src/terralib.lua:1124: Errors reported during typechecking.
/home/manolis/legion/language/src/regent/codegen.t:9614: definition of variable with symbol $is is not in scope in this context
          var domain = c.legion_index_space_get_domain([cx.runtime], [is.index_space])
                                                                       ^
```

**Comment by magnatelee** [comment]
Should be fixed by 9439d818.

**Comment by manopapad** [comment]
Fixed
